src/Main.hs

```diff
@@ -0,0 +1,198 @@
+import qualified Dhall.Core as Dhall
+import qualified Dhall.Map as Map
+import qualified Data.Map.Strict as Data.Map
+import qualified Data.Set as Set
+import qualified Data.Text as Text
+import qualified Data.List as List
+import qualified Dhall.Parser as Dhall
+import qualified Data.Text.Prettyprint.Doc             as Pretty
+import qualified Data.Text.Prettyprint.Doc.Render.Text as PrettyText
+
+import Data.Bifunctor (first, second)
+import GHC.Generics (Generic)
+import Data.Maybe
+import Data.Aeson
+import Data.Text (Text)
+import Data.Set (Set)
+import Data.Map (Map)
+import Control.Applicative (optional)
+import Control.Monad (join)
+
+data BaseData = BaseData
+  { kind       :: Text
+  , apiVersion :: Text
+  } deriving (Generic, Show)
+
+instance FromJSON BaseData where
+  parseJSON = withArray "array of values" $ \arr -> withObject "baseData" (\o -> do
+    group   <- o .:? "group" .!= ""
+    kind    <- o .: "kind"
+    version <- o .: "version"
+    let apiVersion = group <> "/" <> version
+    pure BaseData{..})
+    (head $ Vector.toList arr)
+
+
+data Definition = Definition
+  { typ         :: Maybe Text
+  , ref         :: Maybe Ref
+  , format      :: Maybe Text
+  , description :: Maybe Text
+  , items       :: Maybe Definition
+  , properties  :: Maybe (Map ModelName Definition)
+  , required    :: Maybe (Set FieldName)
+  , baseData    :: Maybe BaseData
+  } deriving (Generic, Show)
+
+instance FromJSON Definition where
+  parseJSON = withObject "definition" $ \o -> do
+    typ         <- o .:? "type"
+    ref         <- o .:? "$ref"
+    format      <- o .:? "format"
+    properties  <- o .:? "properties"
+    required    <- o .:? "required"
+    items       <- o .:? "items"
+    description <- o .:? "description"
+    baseData    <- fmap join $ optional (o .:? "x-kubernetes-group-version-kind")
+    pure Definition{..}
+
+newtype Ref = Ref { unRef :: Text }
+  deriving (Generic, Show, FromJSON)
+
+newtype ModelName = ModelName { unModelName :: Text }
+  deriving (Generic, Show, FromJSON, FromJSONKey, Ord, Eq, Pretty.Pretty)
+
+newtype FieldName = FieldName { unFieldName :: Text }
+  deriving (Generic, Show, FromJSON, FromJSONKey, Ord, Eq, Pretty.Pretty)
+
+data Swagger = Swagger
+  { definitions :: Map ModelName Definition
+  } deriving (Generic, Show)
+
+instance FromJSON Swagger
+
+type Expr = Dhall.Expr Dhall.Src Dhall.Import
+
+
+getSwagger :: IO Swagger
+getSwagger = do
+  -- Fetch and parse the Swagger spec
+  req <- HTTP.parseRequest swaggerUrl
+  res <- HTTP.httpJSON req
+  pure $ HTTP.getResponseBody res
+  where
+    kubernetesTag = "v1.13.4"
+
+    swaggerUrl = "https://raw.githubusercontent.com/kubernetes/kubernetes/"
+      <> kubernetesTag
+      <> "/api/openapi-spec/swagger.json"
+
+-- | Get all the required fields for a model
+--   See https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields
+--   TLDR: because k8s API allows PUTS etc with partial data,
```

Comment:
I wonder if we could  poke upstream about this to improve this situation? There must be others running into this issue as well. Worth a shot I guess

Reply:
+1 on this

Reply:
I just realized that we already have an issue to track this: #8